framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -45,6 +45,10 @@ MeshDiagnosticsGenerator::validParams()
       "check_for_watertight_sidesets",
       chk_option,
       "whether to check for external sides that are not assigned to any sidesets");
+  params.addParam<MooseEnum>(
+      "check_for_watertight_nodesets",
+      chk_option,
+      "whether to check for external nodes that are not assigned to any nodeset");
   params.addParam<MooseEnum>(
```

Comment:
let s add a parameter for the nodesets that are supposed to englobe the entire mesh. when this parameter is in use, the nodes are only checked against those "blessed" nodesets

same for sidsets
actually now that I see that this is not an option for sidesets, let s keep it for future work

Reply:
Sounds good. I'll open a new issue for adding that parameter to the sideset and nodeset checks.